File: code/jvm/infrastructure/src/main/java/pt/isel/infrastructure/security/SpringPasswordEncoderAdapter.java
package pt.isel.infrastructure.security;

import org.springframework.stereotype.Component;
import pt.isel.domain.security.PasswordEncoder;

@Component
public class SpringPasswordEncoderAdapter implements PasswordEncoder {

    private final org.springframework.security.crypto.password.PasswordEncoder springEncoder;

    public SpringPasswordEncoderAdapter(org.springframework.security.crypto.password.PasswordEncoder springEncoder) {
        this.springEncoder = springEncoder;
    }

    @Override
    public String encode(String rawPassword) {
        return springEncoder.encode(rawPassword);
    }

    @Override
    public boolean matches(String rawPassword, String encodedPassword) {
        return springEncoder.matches(rawPassword, encodedPassword);
    }
}

File: code/jvm/infrastructure/src/main/kotlin/pt/isel/infrastructure/SpringPasswordEncoderAdapter.kt
package pt.isel.infrastructure

import org.springframework.stereotype.Component
import pt.isel.domain.auth.PasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder as SpringPasswordEncoder

@Component
class SpringPasswordEncoderAdapter(
    private val springEncoder: SpringPasswordEncoder,
) : PasswordEncoder {
    override fun encode(rawPassword: String): String = springEncoder.encode(rawPassword)

    override fun matches(
        rawPassword: String,
        encodedPassword: String,
    ): Boolean = springEncoder.matches(rawPassword, encodedPassword)
}
